src/mean.jl

```diff
@@ -29,99 +9,139 @@ julia> srs = SimpleRandomSample(apisrs;popsize=:fpc);
 
 julia> mean(:enroll, srs)
 1×2 DataFrame
- Row │ mean     sem     
+ Row │ mean     SE     
      │ Float64  Float64 
 ─────┼──────────────────
    1 │  584.61  27.3684
 ```
 """
 function mean(x::Symbol, design::SimpleRandomSample)
+    function se(x::Symbol, design::SimpleRandomSample)
+        variance = design.fpc * Statistics.var(design.data[!, x]) / design.sampsize 
+        return sqrt(variance)
+    end
     if isa(design.data[!, x], CategoricalArray)
         gdf = groupby(design.data, x)
         p = combine(gdf, nrow => :counts)
         p.mean = p.counts ./ sum(p.counts)
         # variance of proportion
         p.var = design.fpc .* p.mean .* (1 .- p.mean) ./ (design.sampsize - 1)
-        p.sem = sqrt.(p.var)
+        p.se = sqrt.(p.var)
         return select(p, Not([:counts, :var]))
     end
-    return DataFrame(mean=mean(design.data[!, x]), sem=sem(x, design))
+    return DataFrame(mean=mean(design.data[!, x]), SE=se(x, design))
 end
 
+"""
+    mean(x, design)
+Estimate the population mean of a variable of a simple random sample, and the corresponding standard error.
+
+```jldoctest
+julia> apisrs = load_data("apisrs");
+
+julia> srs = SimpleRandomSample(apisrs;popsize=:fpc);
+
+julia> mean([:api00, :api99], srs)
+2×3 DataFrame
+ Row │ names   mean     SE     
+     │ String  Float64  Float64 
+─────┼──────────────────────────
+   1 │ api00   656.585  9.24972
+   2 │ api99   624.685  9.5003
+```
+"""
 function mean(x::Vector{Symbol}, design::SimpleRandomSample)
-    means_list = []
-    for i in x
-        push!(means_list, mean(i, design))
-    end
-    df = reduce(vcat, means_list)
+    df = reduce(vcat, [mean(i, design) for i in x])
     insertcols!(df, 1, :names => String.(x))
     return df
 end
 
 """
-Inner method for `by`
-"""
-# Inner methods for `by`
-function sem_by(x::AbstractVector, design::SimpleRandomSample)
-    # domain size
-    dsize = length(x)
-    # sample size
-    ssize = design.sampsize
-    # fpc
-    fpc = design.fpc
-    # variance of the mean
-    variance = (dsize / ssize)^(-2) / ssize * fpc * ((dsize - 1) / (ssize - 1)) * var(x)
-    # return the standard error
-    return sqrt(variance)
-end
+Estimate domain-wise mean. 
+```jldoctest
+julia> using Survey; 
 
+julia> srs = load_data("apisrs"); 
+
+julia> srs = SimpleRandomSample(srs; popsize = :fpc);
+
+julia> mean(:api00, :cname, srs) |> first
```

Comment:
I think it's better to leave the entire resulting data frame, not just the first row. It's more clear what `mean` does in this case.

Reply:
Can you commit to https://github.com/ayushpatnaikgit/Survey.jl/tree/cleaning
?